plugins/org.yakindu.sct.model.sexec/src/org/yakindu/sct/model/sexec/transformation/ng/ReactMethod.xtend

```diff
@@ -0,0 +1,368 @@
+import org.yakindu.base.types.Parameter
+import org.yakindu.base.types.Property
+import org.yakindu.base.types.TypedElement
+import org.yakindu.base.types.TypesFactory
+import org.yakindu.base.types.typesystem.ITypeSystem
+import org.yakindu.sct.model.sexec.Check
+import org.yakindu.sct.model.sexec.ExecutionFlow
+import org.yakindu.sct.model.sexec.ExecutionNode
+import org.yakindu.sct.model.sexec.ExecutionState
+import org.yakindu.sct.model.sexec.If
+import org.yakindu.sct.model.sexec.LocalVariableDefinition
+import org.yakindu.sct.model.sexec.Method
+import org.yakindu.sct.model.sexec.Return
+import org.yakindu.sct.model.sexec.Sequence
+import org.yakindu.sct.model.sexec.StateVector
+import org.yakindu.sct.model.sexec.Step
+import org.yakindu.sct.model.sexec.transformation.SexecElementMapping
+import org.yakindu.sct.model.sexec.transformation.SexecExtensions
+import org.yakindu.sct.model.sexec.transformation.SgraphExtensions
+import org.yakindu.sct.model.sgraph.RegularState
+import org.yakindu.sct.model.sgraph.Statechart
+
+/**
+ * React method is an artifact concepts that is created for each state machine state and the statechart
+ * itself. It defines the react behavior for each reactive element of the state machine. 
+ * 
+ * @author Axel Terfloth - terfloth@itemis.de
+ *  
+ */
+class ReactMethod {
+	
+	@Inject extension SexecElementMapping mapping
+	@Inject extension SexecExtensions sexec
+	@Inject extension SgraphExtensions sgraph
+	@Inject extension ITypeSystem typeSystem
+	
+	
+	/**
+	 * Declares the react methods for all ExecutionNode objects. This are the ExecutionStates and the ExecutionFlow itself.
+	 */	
+	def declareReactMethods(ExecutionFlow it) {
+
+		flow.declareReactMethod		
+		states.forEach[s | s.declareReactMethod ]
+	}
+	
+	/**
+	 * Define the react methods for all ExecutionNode objects. This are the ExecutionStates and the ExecutionFlow itself.
+	 */	
+	def defineReactMethods(ExecutionFlow it) {
+
+		flow.defineReactMethod
+		states.forEach[s | s.defineReactMethod ]
+		
+	}
+	
+	def dispatch ExecutionNode declareReactMethod(RegularState state) {
+		state.create.declareReactMethod
+	}	
+	
+
+	def dispatch ExecutionNode declareReactMethod(Statechart statechart) {	
+		statechart.flow.declareReactMethod 
+	}	
+
+	def dispatch ExecutionNode declareReactMethod(ExecutionNode node) {
+		node => [
+			features.add( sexecFactory.createMethod => [ m |
+				m.name = "react"
+				m._type(_bool)
+				m._param("try_transition", _bool)
+			])
+		]
+	}
+
+	
+	def defineReactMethod(ExecutionFlow it) {
+		reactMethod => [ body = 
+			_sequence(
+				flow.createLocalReactionSequence,
+				_return(_false)	
+			) => [ comment = "State machine reactions."]	
+		]		
+	}
+	
+	
+	
+	def defineReactMethod(ExecutionState execState) {
+		
+		val state = execState.sourceElement as RegularState
+		val reactMethod = execState.reactMethod
+		val childFirst = state.statechart.isChildFirstExecution
+				
+		val shouldProcessParent = 
+			if (! childFirst) 
```

Comment:
I'm not sure how this works. Where does `sv` actually come from? Isn't it null?

Reply:
Got it, fancy! But maybe it's a tad too fancy to use it only once.

Reply:
Reviews should be fun & educational ;-)

Reply:
The construct was copied from the original sequence construction (ReactionBuilder) where it is used at multiple times. 

Inlined it into the processParent expression.